django_project/geohosting/serializer/product.py

```diff
@@ -34,9 +34,19 @@ class ProductDetailSerializer(serializers.ModelSerializer):
     packages = serializers.SerializerMethodField()
     product_meta = serializers.SerializerMethodField()
 
+    def __init__(self, *args, **kwargs):
+        self.currency = kwargs.pop('currency', None)
+        super().__init__(*args, **kwargs)
+
     def get_packages(self, obj: Product):
         packages = obj.packages.all()
-        preferred_currency_order = ['EUR', 'USD', 'ZAR']
+
+        preferred_currency_order = (
```

Comment:
We can delete preferred currency order as we already know what currency should be returned
And also we can delete 
```
unique_packages = {}
```
code, as we just return package with correct currency